zerver/lib/emoji_utils.py

```diff
@@ -0,0 +1,25 @@
+# This file doesn't import from django so that we can use it in `build_emoji`
+
+
+def unqualify_emoji(emoji: str) -> str:
+    # Starting from version 4.0.0, `emoji_datasource` package has started to
+    # add an emoji presentation variation selector for certain emojis which
+    # have defined variation sequences. The emoji presentation selector
+    # "qualifies" an emoji, and an "unqualified" version of an emoji does
+    # not have an emoji presentation selector.
+    #
+    # Since in informal environments(like texting and chat), it is more
+    # appropriate for an emoji to have a colorful display so until emoji
+    # characters have a text presentation selector, it should have a
+    # colorful display. Hence we can continue using emoji characters
+    # without appending emoji presentation selector.
+    # (http://unicode.org/reports/tr51/index.html#Presentation_Style)
+    return emoji.replace("\ufe0f", "")
+
+
+def emoji_to_hex_codepoint(emoji: str) -> str:
+    return "-".join(f"{ord(c):04x}" for c in emoji)
+
+
+def hex_codepoint_to_emoji(hex: str) -> str:
+    return "".join(chr(int(h, 16)) for h in hex.split("-"))
```

Comment:
This could also be used in `zerver.lib.push_notifications.get_mobile_push_content`.

```diff
             match = re.search(r"emoji-(?P<emoji_code>\S+)", classes)
             if match:
                 emoji_code = match.group("emoji_code")
-                char_repr = ""
-                for codepoint in emoji_code.split("-"):
-                    char_repr += chr(int(codepoint, 16))
-                return char_repr
+                return hex_codepoint_to_emoji(emoji_code)
```

Reply:
Oh and also in `generate_emoji_names`.

```diff
             # create the unicode character(s) for the emoji, since this is the key into the CLDR data
-            emoji = "".join(
-                chr(int(h, 16))
-                for h in (emoji_dict["non_qualified"] or emoji_dict["unified"]).split("-")
-                # We have to do this check because of this upstream bug:
-                # https://github.com/iamcal/emoji-data/pull/217
-                if h != "FE0F"
-            )
+            # We can't just use emoji_dict["non_qualified"] because of this upstream bug:
+            # https://github.com/iamcal/emoji-data/pull/217
+            emoji = unqualify_emoji(hex_codepoint_to_emoji(emoji_dict["unified"]))
             if emoji not in CLDR_DATA:
```